pkg/cmd/grafana/apiserver/cmd.go

```diff
@@ -73,15 +73,15 @@ func newCommandStartExampleAPIServer(o *APIServerOptions, stopCh <-chan struct{}
 				return err
 			}
 
-			config, err := o.Config()
-			if err != nil {
-				return err
-			}
-
 			if o.Options.TracingOptions.TracingService != nil {
 				tracer.InitTracer(o.Options.TracingOptions.TracingService)
 			}
 
+			config, err := o.Config(tracer)
```

Comment:
Is this ok @marefr? Just making sure the ordering isn't too disturbed here - I needed to pass in the `tracer` to `Config()`.

Reply:
Yes I think so 👍

Reply:
I broke tracing here.

`tracer.InitTracer(o.Options.TracingOptions.TracingService)` depends on `o.Config()`

Reply:
Sorry about that. So I think that you rather should do `config, err := o.Config()` and then in server.go:

```go
o.factory.GetOptionalMiddlewares(o.Options.TracingOptions.TracingService)...,
```

Reply:
Or call `config, err := o.Config(tracer)` before tracer.InitTracer(o.Options.TracingOptions.TracingService).